src/components/directory-tree-view.js

```diff
@@ -129,55 +154,120 @@ const DirectoryTreeView = ({ rootDirectory }) => {
                 }
                 endIcon={<ChevronRightIcon />}
             >
-                {Array.isArray(node.children)
-                    ? node.children.map((node) => renderTree(node))
+                {Array.isArray(mapData[node.elementUuid].children)
+                    ? mapData[node.elementUuid].children.map((node) =>
+                          renderTree(node)
+                      )
                     : null}
             </TreeItem>
         );
     };
 
     const handleSelect = (nodeId, toggle) => {
         dispatch(setSelectedDirectory(nodeId));
-        fetchDirectoryContent(nodeId).then((childrenToBeInserted) => {
-            dispatch(
-                setCurrentChildren(
+        updateDirectoryChildren(nodeId, toggle);
+    };
+
+    const removeElement = useCallback(
+        (nodeId) => {
+            let expandedCopy = [...expandedRef.current];
+            for (let i = 0; i < expandedCopy.length; i++) {
+                if (expandedCopy[i] === nodeId) {
+                    expandedCopy.splice(i, 1);
+                }
+            }
+            setExpanded(expandedCopy);
+        },
+        [expandedRef]
+    );
+
+    const addElement = useCallback(
```

Comment:
Seems like there still have a bug with the expanded property of the tree view items